pyhap/characteristic.py

```diff
@@ -114,6 +114,11 @@ def __init__(self, display_name, type_id, properties, value=None, broker=None):
         self.setter_callback = None
         self.hap_template = self._create_hap_template()
 
+    def __repr__(self):
+        """Return the representation of the characteristic."""
+        return "<characteristic display_name='{}' value={} properties={}>" \
+            .format(self.display_name, self.display_name, self.properties)
```

Comment:
The second item should be `self.value`. Again the ticks (I get that you want to handle spaces in the display name, but maybe they could also prop in the value or properties)